tests/python/test_python_frontend.py

```diff
@@ -4584,6 +4584,303 @@ def fusion_func(
                     )
                 torch.testing.assert_close(nvf_out[0], ref_out)
 
+    @unittest.skipIf(is_pre_ampere(), "Only supported on Ampere and newer devices.")
+    def test_sdpa_bwd(self):
+        N, H, L, S, E = 4, 8, 16, 16, 8
+
+        grad_output = torch.randn((N, H, L, E), dtype=torch.bfloat16, device="cuda:0")
+        q = torch.randn((N, H, L, E), dtype=torch.bfloat16, device="cuda:0")
+        k = torch.randn((N, H, S, E), dtype=torch.bfloat16, device="cuda:0")
+        v = torch.randn((N, H, S, E), dtype=torch.bfloat16, device="cuda:0")
+
+        dropout_vals = [None, 0.0, 0.2]
+        is_causal_vals = [None, True, False]
+        scale_vals = [None, 1 / E**0.5, 1e-3]
+
+        def fusion_func(
+            fd: FusionDefinition, has_dropout: bool, has_causal: bool, has_scale: bool
+        ):
+            grad_output = fd.define_tensor(
```

Comment:
Yes, I thought this might be more clear to users in understanding what the different inputs look like -- given that two of them are CPU scalars as well.